src/main/java/org/folio/edge/sip2/MainVerticle.java

```diff
@@ -83,6 +95,20 @@ public void start(Future<Void> startFuture) {
       handlers.put(END_PATRON_SESSION, injector.getInstance(EndPatronSessionHandler.class));
     }
 
+    JsonObject crOptionsJson = config().getJsonObject("tenantConfigRetrieverOptions");
+    ConfigRetrieverOptions crOptions = new ConfigRetrieverOptions(crOptionsJson);
+    ConfigRetriever configRetriever = ConfigRetriever.create(vertx, crOptions);
+
+    configRetriever.getConfig(ar -> {
+      multiTenantConfig = ar.result();
+      log.info("Tenant config loaded: {}", () -> multiTenantConfig.encodePrettily());
+    });
+    
+    configRetriever.listen(change -> {
+      multiTenantConfig = change.getNewConfiguration();
```

Comment:
I don't think thread safety will be a problem here, but I just want to confirm that swapping out the config while other threads are running will be OK.

Reply:
I agree thread safety is most likely not an issue as there is a single thread writing the multiTenantConfig and multiple threads reading it. Writing/setting of the multiTenantConfig happens in one fell swoop rather than piece meal so reader threads won't get corrupt data. The only potential for errors is if a kiosk is removed from config while it is already connected; that connection will not be closed.